### PR TITLE
:lipstick::sparkles:Create the booklets page hero section #15

### LINKED ISSUE #15: 💄 🎨 Create the booklets page hero section
#  Create the booklets page hero section 
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want a nice hero section on the booklets page that captures my attention and introduces the page content. Reuse the text-content component created in #20, and the reusable slider button in  #24. I want a good experience on the web and phone.

Design:

booklets_page_hero

Link to full design:

```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528/
```
 
### Technical Analysis

This feature can be found in libs/features/pages/booklets-page inside a components folder. The component's name is booklets-hero-section.

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 30 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the global style.scss
5. The design is responsive for screen, tablet and phone

### PR DESCRIPTION
**Pull request by darlene9497**
# Description

This pull request introduces the booklets-hero-section component, which reuses the text-section and app-header components. The booklets-hero-section serves as a hero section on the Booklets page, capturing the user's attention and providing an introduction to the page's content.

Fixes #15 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshot
Screenshot from 2023-09-08 12-58-07

Tablet view
Screenshot from 2023-09-08 12-42-42

Mobile view
Screenshot from 2023-09-08 13-02-27

## How Has This Been Tested?

- [x] Test A - The server runs successfully without generating any warnings

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/pages/booklets-page/src/lib/booklets-page.module.ts [hunk 1]
```diff
@@ -1,10 +1,16 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
+
 import { BookletsPageComponent } from './main/booklets-page/booklets-page.component';
+import { BookletsHeroSectionComponent } from './components/booklets-hero-section/booklets-hero-section.component';
+import { AppHeaderModule } from '@elewa-website/elements/layout/header';
+import { TextsModule } from '@elewa-website/elements/layout/texts';
+import { ButtonsModule } from '@elewa-website/elements/layout/buttons';
+
 import { BookletsRoutingModule } from './booklets.routing';
```
Comment (AnnaL001) [inline libs/features/pages/booklets-page/src/lib/booklets-page.module.ts:11]: Kindly, check the order of imports. The elements imports should appear before local imports

File: libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export class BookletsHeroSectionComponent {
+  textData = {
+    title: 'Booklets',
+    subtitle: '',
+    description: 'Your organisation has the potential to turn every task into an opportunity to learn. From growing crops, to nursing someone to good health, to developing the next-generation technology, let’s work together to make the magic happen.'
+  };
+  heroButtonData = {
+    text: 'Unlock your potential'
+  }
+}
```
Comment (AnnaL001) [inline libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.ts:17]: To keep the component lean and ensure readability and maintainability, the 'textData' and 'heroButtonData' variable content can be moved to a Typescript file within the data module and referenced accordingly.

**Review by AnnaL001 — CHANGES_REQUESTED** [review]
Nicely done, just a number of changes before this is merged

File: libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.ts (hunk 2 above)
Comment (AnnaL001) [inline libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.ts:17]: To keep the component lean and ensure readability and maintainability, the 'textData' and 'heroButtonData' variable content can be moved to a Typescript file within the data module and referenced accordingly.

File: libs/features/pages/booklets-page/src/lib/booklets-page.module.ts (hunk 1 above)
Comment (darlene9497) [reply libs/features/pages/booklets-page/src/lib/booklets-page.module.ts:12]: @AnnaL001 I have made the changes

File: libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (darlene9497) [reply libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.ts]: @AnnaL001 I have made the necessary changes

**Review by AnnaL001 — DISMISSED** [review]

File: libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.scss [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        background: transparent linear-gradient(111deg, var(--elewa-website-booklets-hero-section-curve-color) 0%, transparent 100%) 0% 0% no-repeat padding-box;
+        opacity: 0.87;
+        clip-path: polygon(
+            0% 100%, 100% 100%, 100% 20%, 97% 16%, 94% 12%, 92% 10%, 90% 8%, 88% 6%, 86% 4%, 84% 3%, 82% 2%, 80% 1%, 78% 1%, 76% 0%, 74% 1%, 72% 1%, 70% 2%, 68% 3%, 66% 4%, 64% 6%, 62% 8%, 60% 10%, 58% 12%, 56% 16%, 54% 20%, 52% 22%, 50% 26%, 48% 30%, 46% 34%, 44% 38%, 42% 42%, 40% 46%, 38% 50%, 36% 54%, 34% 58%, 32% 62%, 30% 66%, 28% 70%, 26% 74%, 24% 78%, 22% 80%, 20% 84%, 18% 88%, 16% 92%, 14% 94%, 12% 96%, 10% 97%, 8% 98%, 6% 99%, 4% 99%, 2% 100%, 0% 100%
+        );
+        shape-outside: polygon(
+            0% 100%, 100% 100%, 100% 20%, 97% 16%, 94% 12%, 92% 10%, 90% 8%, 88% 6%, 86% 4%, 84% 3%, 82% 2%, 80% 1%, 78% 1%, 76% 0%, 74% 1%, 72% 1%, 70% 2%, 68% 3%, 66% 4%, 64% 6%, 62% 8%, 60% 10%, 58% 12%, 56% 16%, 54% 20%, 52% 22%, 50% 26%, 48% 30%, 46% 34%, 44% 38%, 42% 42%, 40% 46%, 38% 50%, 36% 54%, 34% 58%, 32% 62%, 30% 66%, 28% 70%, 26% 74%, 24% 78%, 22% 80%, 20% 84%, 18% 88%, 16% 92%, 14% 94%, 12% 96%, 10% 97%, 8% 98%, 6% 99%, 4% 99%, 2% 100%, 0% 100%
```
Comment (LemmyMwaura) [inline libs/features/pages/booklets-page/src/lib/components/booklets-hero-section/booklets-hero-section.component.scss:33]: This creates a very off background. All you need is to create a smooth pill-shaped rectangle and then transform it then cut out the overflow.

**Review by LemmyMwaura — APPROVED** [review]

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**